File: src/main/java/org/amithalpert/Player.java
package org.amithalpert;
import java.awt.*;
import java.awt.event.KeyEvent;
import java.awt.event.KeyListener;
import javax.swing.*;

public class Player extends Thread implements KeyListener {
	GameScreen panel;

	private int playerWidth = 130, playerHeight = 115;
	private int playerX = 800, playerY;
	private int velX = 0, velY = 0;
	Rectangle hitBox = new Rectangle(0, 0, playerWidth, playerHeight);
	Image Texture;


	public Player(GameScreen panel)
	{
		this.panel=panel;
		Texture = new ImageIcon("src/main/resources/nix.png").getImage();

		start();
	}
	
	public void run()
	{
		while(true)
		{


			updatePlayer();
			velY  += 1.0;


		   try {
			   	Thread.sleep(20);
		   	   }catch (InterruptedException e) {}
				
			panel.repaint();
		}
	}

	private void updatePlayer() {

		//Updates X AND Y Position of the player
		playerX += velX;
		playerY += velY;


		hitBox.setLocation(playerY, playerX);
	}
	
	
	
	public void draw(Graphics g)
	{
		g.drawImage(Texture, playerX,  playerY, playerWidth, playerHeight,null);
	}

	@Override
	public void keyTyped(KeyEvent key) {

	}

	@Override
	public void keyPressed(KeyEvent key) {
		if (key != null) {
			switch (key.getKeyCode()){
				case KeyEvent.VK_SPACE:
					velY -= 45.0;
					break;
			}
		}

	}

	@Override
	public void keyReleased(KeyEvent key) {

	}
}

File: src/main/kotlin/Player.kt
import java.awt.Graphics
import java.awt.Image
import java.awt.event.KeyEvent
import java.awt.event.KeyListener
import java.awt.geom.Rectangle2D
import javax.swing.ImageIcon

class Player(var panel: GameScreen) : Thread(), KeyListener{

    // constants
    private val playerWidth = 130
    private val playerHeight = 115
    private val gravityAmount = 300
    // player characteristics
    private var playerX : Double = 800.0
    private var playerY : Double = 0.0
    private var velX : Double = 0.0
    private var velY : Double = 0.0
    private var playerHitBox : Rectangle2D? = null
    // animations
    private var ballImage: Image



    init {
        val img = ImageIcon("src/main/resources/nix.png")
        ballImage = img.image
        playerHitBox?.setRect(32.0,34.0,54.0,54.0)


        start()
    }



    override fun run() {
        while (true) {


            updatePlayer()


            try {
                sleep(10)
            } catch (e: InterruptedException) {
            }
            panel.repaint()
        }
    }

    fun updatePlayer() {

        //Updates X AND Y Position of the player
        playerX += velX
        playerY += velY

    }

    fun draw(g: Graphics) {
        g.drawImage(ballImage, playerX.toInt(), playerY.toInt(), playerWidth, playerHeight, null)
    }

    override fun keyTyped(p0: KeyEvent?) {

    }

    override fun keyPressed(p0: KeyEvent?) {
        println("deezsssssssssssssssssss")
    }

    override fun keyReleased(p0: KeyEvent?) {

    }
}